File: Python/7py 3/title_to_number.py
"""
|--------------------------------------------------------------------------
| Excel sheet column numbers
|--------------------------------------------------------------------------
|
"""

def title_to_number(title):
   res = 0
   for char in title:
      res = res * 26 + (ord(char) - ord('A') + 1)
   return res


def title_to_number(title):
   return sum((ord(char) - 64) * 26 ** i for i, char in enumerate(title[::-1]))


from functools import reduce
def title_to_number(s):
   return reduce(lambda a,c: a*26+ord(c)-64, s, 0)


print(title_to_number('A'),1)
print(title_to_number('Z'),26)
print(title_to_number('AA'),27)
print(title_to_number('AZ'),52)
print(title_to_number('BA'),53)
print(title_to_number('CODEWARS'),28779382963)

File: PHP/7php 3/titleToNumber.php
<?php
/*
|--------------------------------------------------------------------------
| Excel sheet column numbers
|--------------------------------------------------------------------------
|
*/

function titleToNumber($title) {
   $res = 0;
   for ($i = 0; $i < strlen($title); $i++) {
      $res = $res * 26 + (ord($title[$i]) - ord('A') + 1);
   }
   return $res;
}


function titleToNumber($title) {
   return array_reduce(str_split($title), fn ($c, $i) => $c * 26 + (ord($i) - 64));
}


function titleToNumber($title) {
   $res = $i = 0;
   foreach (array_reverse(str_split($title)) as $t) $res += (ord($t)-64) * 26**$i++; 
   return $res;
}


print_r(titleToNumber("A") . ' ' . 1 . '
');
print_r(titleToNumber("Z") . ' ' . 26 . '
');
print_r(titleToNumber("AA") . ' ' . 27 . '
');
print_r(titleToNumber("AZ") . ' ' . 52 . '
');
print_r(titleToNumber("BA") . ' ' . 53 . '
');
print_r(titleToNumber("CODEWARS") . ' ' . 28779382963 . '
');